ammo_check/gamedata/scripts/ammo_check_mcm.script

```diff
@@ -205,14 +205,16 @@ function check_Ammo()
     local sec = weapon:section()
     local top_round = nil
 
+    local mag_data = get_data(weaponId)
+
     if (is_jammed_weapon(weapon)) then
         message = gc("st_ac_jammed")
         clr = use_clr and clr00_Red or nil
     elseif weapon:weapon_in_grenade_mode() then
         message = currentAmmo == 1 and gc("st_ac_grenade") or gc("st_ac_empty")
         clr = use_clr and clr03_Amber
     elseif currentAmmo == 0 then
-        if is_supported_weapon(sec) and get_data(weaponId) == nil then
+        if is_supported_weapon(sec) and not mag_data then
```

Comment:
isn't the point of this change that a supported weapon will always have mag_data?

Reply:
or was this check some sort of fringe error case and not something that was ever suposed to be shown?